### DISCUSSION TITLE
Lesson 8 Different hex

### DISCUSSION THREAD

**Discussion by 0xSecurious** [Cyfrin/foundry-full-course-cu#1994] [Q&A] [answered]
When calling fund() function, in my metamask i get this hex:

![image](https://github.com/Cyfrin/foundry-full-course-cu/assets/141664845/3588e350-48dc-4bb1-836d-1cde5654c804)

Cast sig gives me this hex of "fund()": **0xb60d4288**
It's the same as in the lesson.

Is it different due to passed parameters and how can then verify what function i am calling?

**Comment by EngrPips** [comment, marked as answer]
<img width="1147" alt="Screenshot 2024-07-11 at 18 59 37" src="https://github.com/Cyfrin/foundry-full-course-cu/assets/95760918/de8334ee-f01a-4bb5-b05f-133e644f8365">

I believe the above image does justice to the explanation, you were not passing in the argument type that the `fund` function takes when you did `cast sig` command

**Reply by 0xSecurious** [reply]
Yeah that was the problem, didn't use ```cast sig``` the right way.

Thank you!
